models/classes/session/Business/Service/SessionCookieService.php

```diff
@@ -15,67 +15,103 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ * Copyright (c) 2020-2022 (original work) Open Assessment Technologies SA;
  *
  * @author Sergei Mikhailov <sergei.mikhailov@taotesting.com>
  */
 
-declare(strict_types=1);
+declare (strict_types=1);
 
 namespace oat\tao\model\session\Business\Service;
 
 use common_http_Request as Request;
 use oat\tao\model\service\InjectionAwareService;
 use oat\tao\model\session\Business\Contract\SessionCookieAttributesFactoryInterface;
 use oat\tao\model\session\Business\Contract\SessionCookieServiceInterface;
+use oat\tao\model\session\Business\Domain\SessionCookieAttribute;
 use tao_helpers_Uri as UriHelper;
 
 class SessionCookieService extends InjectionAwareService implements SessionCookieServiceInterface
 {
     /** @var SessionCookieAttributesFactoryInterface */
     private $sessionCookieAttributesFactory;
+    private $attributes = [];
+    private $sessionCookieParams = [];
 
     public function __construct(SessionCookieAttributesFactoryInterface $sessionCookieAttributesFactory)
     {
         parent::__construct();
-
         $this->sessionCookieAttributesFactory = $sessionCookieAttributesFactory;
+        //it will brake the unit test logic
+        //   $this->init();
     }
 
-    public function initializeSessionCookie(): void
+    private function init(): void
     {
+        $iterator = $this->sessionCookieAttributesFactory->create()->getIterator();
+        while ($iterator->valid()) {
+            $this->attributes[] = $iterator->current();
+            $iterator->next();
+        }
```

Comment:
```suggestion
        $attributes = [];
        $iterator = $this->sessionCookieAttributesFactory->create()->getIterator();
        while ($iterator->valid()) {
            $attributes[] = $iterator->current();
            $iterator->next();
        }
```
reason of it mentioned here https://github.com/oat-sa/tao-core/pull/3712/files#r1057772958